### DISCUSSION TITLE
How to combine isometric images?

### DISCUSSION THREAD

**Discussion by temotodochi** [ImageMagick/ImageMagick#4622] [Help] [answered]
I'm lacking correct names and words so my google-fu is failing big time. 
I wish to tile several large (40MB rgba png) isometric map images into one large image, but haven't found the correct command syntax to do it. -adjoin merges them side to side which is not the correct place. 

Here are some examples in smaller size https://imgur.com/a/r9VeMU8

**Comment by fmw42** [comment]
The correct command is -+append not adjoin.  See https://legacy.imagemagick.org/Usage/layers/#append

Append images in rows using +append. Then append all the rows with -append.

**Reply by temotodochi** [reply]
Thanks for the note, but i'm not sure how append works with isometric images that are not joined sideways on. Do i need to smush them? How do i figure out the geometry? I have 8x8 grid of images i'd have to join up.

**Comment by GeeMack** [comment, marked as answer]
To assemble a 8x8 grid of your example input images using ImageMagick v7, the command below should get that result. I used one of your input samples with a transparent background and a few irregularities of trees and buildings along the top edges. I duplicated it 64 times, labeled them with numbers, and named the files "mapNN.png". I resized them to 25% to input for my test, but the command should work exactly the same with your full size 4800 pixel wide images if they fit in your memory. This is the command I used on the 64 1200 pixel wide input images...

    magick map*.png -background none ^
       ( -clone 0-7 -set page +%[fx:t*w/2]+%[fx:t*w/4] -layers merge ) -delete 0-7 ^
       ( -clone 0-7 -set page +%[fx:t*w/2]+%[fx:t*w/4] -layers merge ) -delete 0-7 ^
       ( -clone 0-7 -set page +%[fx:t*w/2]+%[fx:t*w/4] -layers merge ) -delete 0-7 ^
       ( -clone 0-7 -set page +%[fx:t*w/2]+%[fx:t*w/4] -layers merge ) -delete 0-7 ^
       ( -clone 0-7 -set page +%[fx:t*w/2]+%[fx:t*w/4] -layers merge ) -delete 0-7 ^
       ( -clone 0-7 -set page +%[fx:t*w/2]+%[fx:t*w/4] -layers merge ) -delete 0-7 ^
       ( -clone 0-7 -set page +%[fx:t*w/2]+%[fx:t*w/4] -layers merge ) -delete 0-7 ^
       ( -clone 0-7 -set page +%[fx:t*w/2]+%[fx:t*w/4] -layers merge ) -delete 0-7 ^
       -set page -%[fx:t*w/9]+%[fx:t*w/18] -layers merge -resize 10% output.png

The output is resized to 960 wide to make it reasonable to display here. This is the result...

![output](https://user-images.githubusercontent.com/65475279/147152553-781313bd-ea9b-42cb-8553-c53079331272.png)

The command is in Windows CMD syntax. To run it on a *nix system you'll need to at least replace the continued-line carets "^" with backslashes "\\" and escape the parentheses "(...)" with backslashes "\\(...\\)".

**Reply by temotodochi** [reply]
That's brilliant! Thank you very much!
